packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -626,6 +597,63 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	}
 
+	private createRevertible(
+		revision: RevisionTag,
+		kind: CommitKind,
+		checkout: TreeCheckout,
+		onRevertibleDisposed: (revertible: RevertibleAlpha) => void,
```

Comment:
nit: It was good to change this to be a required parameter rather than optional, as we discussed. However, it can still accept `undefined` as a type. If the type is `onRevertibleDisposed: ((revertible: RevertibleAlpha) => void) | undefined`, then the caller _still_ has to pass something in (it's required), but they can pass in the value `undefined` to mean "nothing". I think that's slightly better than making them pass in `() => {}`, since it doesn't allocate a function object. Up to you though, it's a matter of taste - do it the way you prefer.

Reply:
I love these subtle differences you can use to make callers of the function be aware of what they are doing. This is really interesting 🤯 !

I think going along with your way of either making the `onRevertibleDisposed` a call back function or `undefined` to avoid unnecessary allocation of function object not only enforces strong typing but also ensures efficiency. Thanks for the great feedback!